client/state/data-layer/wpcom/user-settings/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { isEmpty, keys, merge, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { decodeEntities } from 'lib/formatting';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { updateUserSettings, clearUnsavedUserSettings } from 'state/user-settings/actions';
+import {
+	USER_SETTINGS_REQUEST,
+	USER_SETTINGS_SAVE,
+} from 'state/action-types';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function fromApi( apiResponse ) {
+	const decodedValues = {
+		display_name: apiResponse.display_name && decodeEntities( apiResponse.display_name ),
+		description: apiResponse.description && decodeEntities( apiResponse.description ),
+		user_URL: apiResponse.user_URL && decodeEntities( apiResponse.user_URL )
+	};
+
+	return merge( {}, apiResponse, decodedValues );
+}
```

Comment:
> And all that for very little benefit.

debatable :wink: we also write all this in JS instead of rendering server-generated HTML for very little benefit…

my main point was a question if we're already depending on these names in other places in Calypso. I consider it _a good thing_ to perform the mapping on ingress and egress which prevents the tight coupling of the API with Calypso and encourages us to do actual validation and explicit data transformation